src/StackExchange.Redis.Extensions.Core/StackExchangeRedisCacheClient.cs

```diff
@@ -628,8 +628,13 @@ public IEnumerable<string> SearchKeys(string pattern)
 
 			foreach (var endpoint in endPoints)
 			{
-				var dbKeys = Database.Multiplexer.GetServer(endpoint).Keys(Database.Database, pattern);
+				var server = Database.Multiplexer.GetServer(endpoint);
+				if (!server.IsConnected || !server.Features.Scan)

```

Comment:
I quite agree with @EugeneKrapivin if there's a problem, whatever it is (included connection problem) the developer/system engineer must be notified about the problem.
Doing this we are hiding a problem.

Maybe we could do whaat you want integrating a logger (CommonLogging? (asp.net core has its own logger wrapper)).
But I'm not convinced so much about that.
@rushfrisby what do you think?